book/line_editor.md

```diff
@@ -0,0 +1,502 @@
+      selected_text_style: green_reverse  # Color style for selected text
+      marker: "| "       # Indicator that appears when the menu is active
+    }
+
+    ...
+  }
+```
+
+By modifying these parameters you can customize the layout of your menu to your liking.
+
+### History menu
+
+The history menu is a handy way to access the editor history. When activating the menu (default `Ctrl+x`)
+the command history is presented in reverse chronological order, making it extremely easy to select a
+previous command.
+
+The history menu can be configured by modifying the next values from the config object
+
+```bash
+  let $config = {
+    ...
+
+    history_config: {
+      page_size: 10      # Number of entries that will presented when activating the menu
+      selector: "!"      # Character to indicate a quick selection in the menu
+      text_style: green  # Text color style for non selected text
+      selected_text_style: green_reverse  # Color style for selected text
+      marker: "? "       # Indicator that appears when the menu is active
+    }
+
+    ...
+  }
+
+```
+
+When the history menu is activated, `page_size` number of records is pulled from the history and presented
+in the menu. If there is space in the terminal, when you press again `Ctrl+x` the menu will pull the same
+number of records and append them to the current page. If it isn't possible to present all the pulled
+records, the menu will create a new page. The pages can be navigated by pressing `Ctrl+z` to go to
+previous page or `Ctrl+x` to go to next page.
+
+#### Searching the history
+
+To search in your history you can start typing key words for the command you are looking for. Once
+the menu is activated, anything that you type will be replaced by the selected command from your history.
+for example, say that you have already typed this
+
+```bash
+let a = ()
+```
+you can place the cursor inside the `()` and activate the menu. You can filter the history by typing
+key words and as soon as you select an entry, the typed words will be replaced
+
+```bash
+let a = (ls | where size > 10MiB)
+```
+
+#### Menu quick selection
+
+Another nice feature of the menu is the ability to quick select something from it. Say you have
+activated your menu and it looks like this
+
+```bash
+>
+0: ls | where size > 10MiB
+1: ls | where size > 20MiB
+2: ls | where size > 30MiB
+3: ls | where size > 40MiB
+```
+Instead of pressing down to select the fourth entry, you can type `!3` and press enter. This will
+insert the selected text in the prompt position, saving you time scrolling down the menu.
+
+Both, history search and quick selection, can be used together. You can activate you menu, do a quick
+search and then quick select using the quick selection character.
+
+It should be mentioned that if you would like to change the quick selection character for something else,
+you can modify the `selector` value from the `history_config` in the `$config` object.
```

Comment:
Could probably remove this sentence, `selector` is already mentioned in the example above and I don't think it will be a common thing to change